addons/crm_iap_lead/models/crm_lead.py

```diff
@@ -1,11 +1,125 @@
+        except InsufficientCreditError:
+            raise UserError("You do not have enough credits to perform this action.")  
+
+    def _make_request(self, domains):
+        """This method will query the endpoint to get the data for the asked (lead.id, domain) pairs"""
+        reveal_account = self.env['iap.account'].get('reveal')
+        dbuuid = self.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        endpoint = self.env['ir.config_parameter'].sudo().get_param('reveal.endpoint', DEFAULT_ENDPOINT) + '/iap/clearbit/1/lead_enrichment_email'
+        params = {
+            'account_token': reveal_account.account_token,
+            'dbuuid': dbuuid,
+            'domains': domains,
+        }
+        return jsonrpc(endpoint, params=params, timeout=300)
+
+    def _create_message_data(self, company_data):
+        log_data = {
+            'message_title': _("Lead enriched based on email address"),
+            'twitter': company_data['twitter'],
+            'description': company_data['description'],
+            'logo': company_data['logo'],
+            'name': company_data['name'],
+            'phone_numbers': company_data['phone_numbers'],
+            'facebook': company_data['facebook'],
+            'linkedin': company_data['linkedin'],
+            'crunchbase': company_data['crunchbase'],
+            'tech': [t.replace('_', ' ').title() for t in company_data['tech']],
+        }
+        timezone = company_data['timezone']
+        if timezone:
+            log_data.update({
+                'timezone': timezone.replace('_', ' ').title(),
+                'timezone_url': company_data['timezone_url'],
+            })
+        return log_data
+
+    def _send_message(self, company_data=False):
+        if company_data:
+            messages_to_post = self._create_message_data(company_data)
+            self.message_post_with_view('crm_iap_lead.lead_message_template', values=messages_to_post, subtype_id=self.env.ref('mail.mt_note').id)
+        else:
+            self.message_post_with_view('crm_iap_lead.lead_message_not_found', subtype_id=self.env.ref('mail.mt_note').id)
+
+    @api.model
+    def _enrich_leads_from_response(self, data_clearbit):
+        """ This method will get the response from the service and enrich the lead accordingly """
+        for lead_id, data in data_clearbit.items():
+            record = self.browse(int(lead_id))
+            if record.exists():
+                if data:
+                    print(data)
+                    street = False
+                    street2 = False
+                    zip_code = False
+                    city = False
+                    country_id = False
+                    state_id = False
+
+                    if not(record.street or record.street2 or record.zip or record.city or record.state_id or record.country_id):
+                        street = ' '.join(filter(False, (data["street_name"], data["street_number"])))
```

Comment:
https://docs.python.org/3/library/functions.html#filter
False is not a function